src/components/sidenav/sidenav.spec.ts

```diff
@@ -0,0 +1,276 @@
+  TestComponentBuilder,
+  beforeEachProviders,
+  beforeEach,
+} from 'angular2/testing';
+import {provide, Component, DebugElement} from 'angular2/core';
+import {By} from 'angular2/platform/browser';
+
+import {MdSidenav, MdSidenavLayout, MD_SIDENAV_DIRECTIVES} from './sidenav';
+import {AsyncTestFn, FunctionWithParamTokens} from 'angular2/testing';
+import {ComponentFixture} from "angular2/testing";
+import {EventEmitter} from "angular2/core";
+import {Predicate} from "angular2/src/facade/collection";
+
+
+function wait(msec: number) {
+  return new Promise<void>(resolve => window.setTimeout(() => resolve(), msec));
+}
+
+
+function waitOnEvent(fixture: ComponentFixture,
+                     by: Predicate<DebugElement>,
+                     propertyName: string) {
+  fixture.detectChanges();
+
+  // Wait for the animation end.
+  return new Promise(resolve => {
+    const component: any = fixture.debugElement.query(by).componentInstance;
+    component[propertyName].subscribe(resolve);
+  });
+}
+
+
+describe('MdSidenav', () => {
+  let builder: TestComponentBuilder;
+
+  beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => { builder = tcb; }));
+
+  describe('methods', () => {
+    it('should be able to open and close', (done: any) => {
+      let testComponent: BasicTestApp;
+      let fixture: ComponentFixture;
+
+      return builder.createAsync(BasicTestApp)
+        .then((f) => {
+          fixture = f;
+          testComponent = fixture.debugElement.componentInstance;
+
+          const openButtonElement = fixture.debugElement.query(By.css('.open'));
+          openButtonElement.nativeElement.click();
+        })
+        .then(() => wait(1))
+        .then(() => {
+          expect(testComponent.openStartCount).toBe(1);
+          expect(testComponent.openCount).toBe(0);
+        })
+        .then(() => waitOnEvent(fixture, By.directive(MdSidenav), 'onOpen'))
+        .then(() => {
+          expect(testComponent.openStartCount).toBe(1);
+          expect(testComponent.openCount).toBe(1);
+          expect(testComponent.closeStartCount).toBe(0);
+          expect(testComponent.closeCount).toBe(0);
+
+          // Close it.
+          const closeButtonElement = fixture.debugElement.query(By.css('.close'));
+          closeButtonElement.nativeElement.click();
+        })
+        .then(() => wait(1))
+        .then(() => {
+          expect(testComponent.openStartCount).toBe(1);
+          expect(testComponent.openCount).toBe(1);
+          expect(testComponent.closeStartCount).toBe(1);
+          expect(testComponent.closeCount).toBe(0);
+        })
+        .then(() => waitOnEvent(fixture, By.directive(MdSidenav), 'onClose'))
+        .then(() => {
+          expect(testComponent.openStartCount).toBe(1);
+          expect(testComponent.openCount).toBe(1);
+          expect(testComponent.closeStartCount).toBe(1);
+          expect(testComponent.closeCount).toBe(1);
+        })
+        .then(done, done.fail);
```

Comment:
Could you also check that the element is actually has the right `visibility` set at the right times?